lib/src/main/scala/spinal/lib/bus/regif/BusIfAdapter/AhbLite3BusInterface.scala

```diff
@@ -18,21 +18,20 @@ case class AhbLite3BusInterface(bus: AhbLite3, sizeMap: SizeMapping, regPre: Str
 
   val askWrite    = bus.HSEL & bus.HTRANS(1) === True &  bus.HWRITE
   val askRead     = bus.HSEL & bus.HTRANS(1) === True & !bus.HWRITE
-  val doWrite     = bus.HREADY & RegNext(askWrite, False)
-  val doRead      = bus.HREADY & askRead
+  val doWrite     = bus.HREADYOUT & RegNext(askWrite, False)
+  val doRead      = bus.HREADYOUT & askRead
```

Comment:
`HREADY` should be used instead of `HREADYOUT` as the transfer can be delayed ("stuck in address phase") by wait states from the previous transfer (during its data phase).

Reply:
at the view of Master  we should see the slave' READYOUT to find is it ready to write or read , right?
earlier we use HREADY which from master it' not reasonable ,  it should be a bug, right?

Reply:
HREADY comes from the multiplexor (section 2.5).

HREADY is driven by the HREADYOUT of the currently selected slave. HREADY is sent to the master and all the slaves (section 4.2).

So master actually reads HREADY, not HREADYOUT, it is just that during a transaction with a slave, HREADY is connected to HREADYOUT, so the behavior is the same unless if two consecutive transactions are not on the same slave.

Master does not have an HREADY-like output (section 2.2) but it can use HTRANS to insert wait states in the middle of a burst with the BUSY value (b01) (section 3.2).

Reply:
> HREADY is driven by the HREADYOUT of the currently selected slave. HREADY is sent to the master and all the slaves (section 4.2).

you right,  The AHB protocol is special, which is a little different from the common handshake bus.